### PR TITLE
[FLINK-3425] FileOutputFormat closes outStream in case of failure

### PR DESCRIPTION
**Pull request by zentol**

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/api/common/io/FileOutputFormat.java [hunk 1]
```diff
@@ -305,6 +305,12 @@ public void tryCleanupOnError() {
 			this.fileCreated = false;
 			
 			try {
+				close();
+			} catch (IOException e) {
+				LOG.error("Could not properly close FileOutputFormat.", e);
```
Comment (StephanEwen) [inline flink-core/src/main/java/org/apache/flink/api/common/io/FileOutputFormat.java]: This should probably `fail()` the test

Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/api/common/io/FileOutputFormat.java]: We can't check for that. We could expose the outputstream, but I couldn't find a way to check whether the stream was actually closed or not.

Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/api/common/io/FileOutputFormat.java]: apart from trying to write to it.

**Comment by StephanEwen** [comment]
Sorry, I confused the diff.

This looks actually good, +1 to merge

**Comment by tillrohrmann** [comment]
Failing test case seems to be unrelated. Thanks for your work @zentol and the review @StephanEwen. Will merge it.

**Pull request closed without merging**